torch/testing/_internal/common_methods_invocations.py

```diff
@@ -6038,7 +6038,8 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
            dtypesIfCPU=all_types_and_complex(),
            dtypesIfCUDA=floating_and_complex_types_and(torch.float16, *[torch.bfloat16] if CUDA11OrLater else []),
            dtypesIfROCM=floating_types_and(torch.half, torch.bfloat16),
-           backward_dtypesIfCUDA=floating_and_complex_types_and(torch.float16),
+           backward_dtypesIfCUDA=floating_and_complex_types_and(torch.float16,
+                                                                *[torch.bfloat16] if SM60OrLater else []),
```

Comment:
I don't think the sm_60 check would be sufficient, as `bfloat16` should have been introduced in CUDA11 (https://docs.nvidia.com/cuda/archive/11.0/cuda-toolkit-release-notes/index.html#cuda-general-new-features), so this could fail on 10 and previous releases. Wouldn't `if CUDA11OrLater` as done for the forward pass also work here?

Reply:
It unfortunately failed in my previous commit which was using `CUDA11OrLater`, see https://app.circleci.com/pipelines/github/pytorch/pytorch/343601/workflows/695dad4b-86d7-49f4-b4c5-e2aa7b1d0bfa/jobs/14465553

Reply:
We may use something like `(CUDA11OrLater and SM60OrLater)`.

Reply:
Did you want to try that change, @xwang233?

Reply:
The skip on line 6079 should be removed once the dtypes are corrected